### PR TITLE
test: (AR-1553) Federated search

### PR DESCRIPTION
**Pull request by dmkasperski**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/AR-1553" title="AR-1553" target="_blank">SubtaskAR-1553</a>  Adapt current user search to support federated search (UI)
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [x] contains a reference JIRA issue number like `SQPIT-764`
  - [ ] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues
This is follow up PR with tests because previous one had conflicts with kalium so we decided to merge things separately to not block others

### Causes (Optional)

_Briefly describe the causes behind the issues. This could be helpful to understand the adopted solutions behind some nasty bugs or complex issues._

### Solutions

_Briefly describe the solutions you have implemented for the issues explained above._

### Dependencies (Optional)

_If there are some other pull requests related to this one (e.g. new releases of frameworks), specify them here._

Needs releases with:

- [ ] GitHub link to other pull request

#### Test Coverage (Optional)

- [ ] I have added automated test to this contribution

#### How to Test

_Briefly describe how this change was tested and if applicable the exact steps taken to verify that it works as expected._

### Notes (Optional)

_Specify here any other facts that you think are important for this issue._

### Attachments (Optional)

_Attachments like images, videos, etc. (drag and drop in the text box)_

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: app/src/test/kotlin/com/wire/android/ui/home/newconversation/NewConversationViewModelTest.kt [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+                .withFailurePublicSearchResponse()
+                .arrange()
+
+            // When
+            viewModel.search("search")
+            advanceTimeBy(501) // 500ms debounce
+
+            // Then
+            assertEquals(
+                viewModel.state.localContactSearchResult.searchResultState, SearchResultState.Failure(2131755295)
```
Comment (dmkasperski) [inline app/src/test/kotlin/com/wire/android/ui/home/newconversation/NewConversationViewModelTest.kt:85]: in the mock Result.Failure.InvalidRequest but here it's SearchResultState.Failure so I assume that `2131755295` is hashcode for InvalidRequest, in ViewModel we handle all failures exactly the same

**Comment by AndroidBob** [comment]
Build [1](https://10.10.124.134/job/wire-android-reloaded/job/PR-701/1/) **failed**.

**Comment by AndroidBob** [comment]
Build [2](https://10.10.124.134/job/wire-android-reloaded/job/PR-701/2/) **failed**.

**Comment by AndroidBob** [comment]
Build [3](https://10.10.124.134/job/wire-android-reloaded/job/PR-701/3/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(3701123)(3701123)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(3701123)(3701123)-dev-release.apk)

File: app/src/test/kotlin/com/wire/android/ui/home/newconversation/NewConversationViewModelTest.kt (hunk 1 above)
Comment (yamilmedina) [reply app/src/test/kotlin/com/wire/android/ui/home/newconversation/NewConversationViewModelTest.kt]: I think, we could do something like: `viewModel.state.localContactSearchResult.searchResultState::class` and compare it to `Result.Failure.InvalidRequest::class` or better, maybe this is possible also:

`assertTrue(viewModel.state.localContactSearchResult.searchResultState is Result.Failure.InvalidRequest)`

Hash codes can vary from jvm to jvm, leading to flakiness.

**Review by yamilmedina — COMMENTED** [review]
I added a comment about assertion of failure instances. Let me know what you think.
Great work with the federated search 👏👏👏

**Comment by AndroidBob** [comment]
Build [4](https://10.10.124.134/job/wire-android-reloaded/job/PR-701/4/) **failed**.

**Comment by AndroidBob** [comment]
Build [6](https://10.10.124.134/job/wire-android-reloaded/job/PR-701/6/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(3742991)(3742991)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(3742991)(3742991)-dev-release.apk)

**Review by yamilmedina — APPROVED** [review]
Great work man! good to have it covered by tests 👏

**Review by trOnk12 — APPROVED** [review]

**Comment by AndroidBob** [comment]
Build [7](https://10.10.124.134/job/wire-android-reloaded/job/PR-701/7/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(3743771)(3743771)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(3743771)(3743771)-dev-release.apk)

**Pull request merged**